src/Auth.php

```diff
@@ -927,7 +943,8 @@ public function login($login_name, $login_password, $noauto = false, $remember_m
                                 $login_password,
                                 $this->user->fields["auths_id"]
                             );
-                            if ($this->ldap_connection !== false && (!$this->auth_succeded && !$this->user_found)) {
+                            if ($this->user_ldap_error === false && (!$this->auth_succeded && !$this->user_found)) {
```

Comment:
Main problem was located here. As long as the connection to the last LDAP server of the list was successful (i.e. bind with root DN was OK), `$this->ldap_connection` was true.
Following logic was resulting in user deletion:
 - try to connect to server where user was seen on last login/sync: connection failure => `$this->ldap_connection = false; $this->user_found = false;`;
 - try to connect to another server: connection OK but user not found => `$this->ldap_connection = \LDAP\Connection; $this->user_found = false;`;
 - result : `$this->ldap_connection !== false && (!$this->auth_succeded && !$this->user_found)` was `true`.